Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/MacroConstants.kt

```diff
@@ -191,21 +162,40 @@ private fun processCodeSnippet(
         }
     }
 
-    visitChildren(translationUnit, visitor)
+    try {
+        visitChildren(translationUnit, visitor)
 
-    if (state != VisitorState.EXPECT_END) {
-        return null
-    }
-    return when (evalResultKind!!) {
-        CXEvalResultKind.CXEval_Int -> IntegerConstantDef(name, type!!, longValue!!)
-        CXEvalResultKind.CXEval_Float -> FloatingConstantDef(name, type!!, doubleValue!!)
-        else -> null
+        if (state != VisitorState.EXPECT_END) {
+            return null
+        }
+
+        val evalResult = evalResultOrNull!!
+        val type = typeOrNull!!
+        val evalResultKind = clang_EvalResult_getKind(evalResult)
+
+        return when (evalResultKind) {
+            CXEvalResultKind.CXEval_Int ->
+                IntegerConstantDef(name, type, clang_EvalResult_getAsLongLong(evalResult))
+
+            CXEvalResultKind.CXEval_Float ->
+                FloatingConstantDef(name, type, clang_EvalResult_getAsDouble(evalResult))
+
+            CXEvalResultKind.CXEval_ObjCStrLiteral,
+            CXEvalResultKind.CXEval_StrLiteral ->
+                StringConstantDef(name, type, clang_EvalResult_getAsStr(evalResult)!!.toKString())
+
+            CXEvalResultKind.CXEval_CFStr,
```

Comment:
Why? From https://chromium.googlesource.com/chromiumos/third_party/clang/+/master/tools/c-index-test/c-index-test.c it seems clang_EvalResult_getAsStr() works on those strings as well.

Reply:
Because I don't know what it is. Do you?

Reply:
https://developer.apple.com/documentation/corefoundation/cfstring?language=objc ?

Reply:
It is not the answer required.
What an expression is treated by libclang as evaluatable with kind `CXEval_CFStr`?